### PR TITLE
Record LSP request aggregation on extension exit

### PR DESCRIPTION
**Pull request by testforstephen**

### CODE REVIEW

**Review by jdneo — DISMISSED** [review]

File: src/daemon/index.ts [hunk 1]
```diff
@@ -103,21 +96,45 @@ async function traceLSPPerformance(javaExt: vscode.Extension<any>) {
...
    }
 
-   lspUsageStats = new LSPUsageStats(javaExtVersion, sampling);
+   // Enable it since redhat.java@1.23.0
+   if (javaExt.exports?.onWillRequestStart) {
+      lspUsageStats = new LSPUsageStats(javaExtVersion, sampling);
+      try {
+         // Load HdrHistogramJS WASM module
+         if (vscode.env.uiKind === vscode.UIKind.Desktop) {
+            hdr.initWebAssemblySync();
```
Comment (akaroml) [inline src/daemon/index.ts:105]: Any specific reason to use the sync version?

File: src/daemon/index.ts [hunk 2]
```diff
@@ -174,7 +191,7 @@ async function traceLSPPerformance(javaExt: vscode.Extension<any>) {
       }
 
       if (traceEvent.resultLength === 0) {
```
Comment (akaroml) [inline src/daemon/index.ts:193]: How about the isEmpty check to also cover empty objects?

**Review by akaroml — COMMENTED** [review]
2 questions.

File: src/daemon/index.ts (hunk 1 above)
Comment (testforstephen) [reply src/daemon/index.ts:105]: This is the recommended approach from the library. On server side, we can use initAsync directly, but for browser, use the async.

https://github.com/HdrHistogram/HdrHistogramJS#boosting-performances-with-webassembly

File: src/daemon/index.ts (hunk 2 above)
Comment (testforstephen) [reply src/daemon/index.ts:193]: resultLength is not the raw response object. It's a processed value parsed from the response.

See
https://github.com/redhat-developer/vscode-java/blob/812613b5923ba27508f0a15a605f4cf80faa2388/src/TracingLanguageClient.ts#L134

**Review by akaroml — APPROVED** [review]
LGTM!

**Pull request merged**